File: two_pointers/1695_maximum_erasure_value/1695_maximum_erasure_value.go
package main

import "fmt"

func maximumUniqueSubarray(nums []int) int {

	count := make([]int, 10001)
	left := 0
	res := nums[0]
	sum := nums[0]
	count[nums[0]]++
	for i := 1; i < len(nums); i++ {
		sum += nums[i]
		count[nums[i]]++
		for count[nums[i]] > 1 {
			count[nums[left]]--
			sum -= nums[left]
			left++
		}
		res = max(res, sum)
	}

	return res
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	nums := []int{4, 2, 4, 5, 6}
	res := maximumUniqueSubarray(nums)
	fmt.Println(res)
}

File: two_pointers/1695_maximum_erasure_value/1695_maximum_erasure_value.py
from typing import List


class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        dic = {}  
        j = 0
        res = 0
        s = 0
        length = len(nums)
        for i in range(length):
            if nums[i] not in dic:
                dic[nums[i]] = 1
            else:
                dic[nums[i]] += 1
            s += nums[i]
            while dic[nums[i]] > 1:
                dic[nums[j]] -= 1
                s -= nums[j]
                j += 1
            res = max(res,s)
        return res if res > s else s



nums = [4,2,4,5,6]
res = Solution().maximumUniqueSubarray(nums)
print(res)